woocommerce/woocommerce
# Add visibility section to pre-publish panel

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds a visibility section to the pre-publish panel.

![Screenshot 2024-02-08 at 09 50 16](https://github.com/woocommerce/woocommerce/assets/1314156/68fa8000-f4d6-4300-ab27-0add31f17a87)

Closes #43705.

**ACs**

```[tasklist]
### Tasks
- [ ] Create the collapsible Visibility section AkVNGImLgSqCObTQ3idVn7-fi-3359_345477
- [ ] Add `Hide in product catalog` checkbox like in the Organization tab
- [ ] Add `Hide from search results` checkbox like in the Organization tab
- [ ] Add `Enable product reviews` checkbox like in the Organization tab
- [ ] Add `Require a password` checkbox like in the Organization tab
- [ ] When the `Require a password` box is checked, the user can enter the password directly in the panel
- [ ] The section and the block in the Organization tab are fully synced
- [ ] The title `Visibility: Public` changes to `Visibility: Hidden` if `Hide in product catalog` or `Hide from search results` as checked
```



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the New product editor and the Prepublish sidebar (product-pre-publish-modal) are enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New
3. Add a product `Name` and press `Add`.
4. The prepublish panel should be visible.
5. Verify that the visibility section is present in the panel.
6. Check the different options and verify they are synced with the options under the `Organization` tab.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
